### PR TITLE
Delete CAST AI provisioned nodes parameter and remove old example

### PR DESCRIPTION
**Pull request by matas-cast**

### CODE REVIEW

File: examples/eks_cluster/variables.tf [hunk 1]
```diff
@@ -21,11 +21,39 @@ variable "aws_secret_access_key" {
...
-
 }
 
 variable "cluster_name" {
   type        = string
   description = "EKS cluster name in AWS account."
 }
 
+# Addresses with whitelisted access to Kubernetes API
+variable "whitelisted_ips" {
```
Comment (varnastadeus) [inline examples/eks_cluster/variables.tf:32]: do we really need this?

File: castai/resource_castai_eks_cluster.go [hunk 2]
```diff
@@ -261,8 +266,16 @@ func resourceCastaiEKSClusterDelete(ctx context.Context, data *schema.ResourceDa
 
 		if clusterResponse.JSON200.CredentialsId != nil && agentStatus != sdk.ClusterAgentStatusDisconnected {
 			log.Printf("[INFO] Disconnecting cluster.")
+			deleteNodes := false
```
Comment (viktorasm) [inline castai/resource_castai_eks_cluster.go:269]: nitpick: all this "get optional bool or default value" logic breaks the flow here, wrapping into helper func would help a lot:

```go
response, err := client.ExternalClusterAPIDisconnectClusterWithResponse(ctx, clusterId, sdk.ExternalClusterAPIDisconnectClusterJSONRequestBody{
	DeleteProvisionedNodes: getOptionalBool(data, FieldEKSClusterDeleteNodesOnDisconnect, false)
})
//...

func getOptionalBool(data, field, defaultValue) {
}
```

File: examples/eks_cluster/variables.tf (hunk 1 above)
Comment (matas-cast) [reply examples/eks_cluster/variables.tf:32]: good point, we used to have this when we had public/private examples. will remove

File: castai/resource_castai_eks_cluster.go (hunk 2 above)
Comment (matas-cast) [reply castai/resource_castai_eks_cluster.go]: good point, fixed

**Review by varnastadeus — APPROVED** [review]

**Pull request merged**